packaging/resource_suite_s3_cache.py

```diff
@@ -40,7 +40,34 @@
 from .resource_suite import ResourceSuite
 from .test_chunkydevtest import ChunkyDevTest
 
+# Used to make files that are not-zero but have random bytes spread throughout
+def make_arbitrary_file(f_name, f_size, buffer_size=32*1024*1024):
+    # do not care about true randomness
+    random.seed(5)
+    bytes_written = 0
+    buffer = buffer_size * [0x78]       # 'x' - bytearray() below appears to require int instead
+                                        #       of char which was valid in python2
+    with open(f_name, "wb") as out:
 
+        while bytes_written < f_size:
+
+            if f_size - bytes_written < buffer_size:
+                to_write = f_size - bytes_written
+                buffer = to_write * [0x78]  # 'x'
+            else:
+                to_write = buffer_size
+
+            current_char = random.randrange(256)
+
+            # just write some random byte each 1024 chars
+            for i in range(0, to_write, 1024):
+                buffer[i] = current_char
+                current_char = random.randrange(256)
+            buffer[len(buffer)-1] = random.randrange(256)
+
+            out.write(bytearray(buffer))
+
+            bytes_written += to_write
```

Comment:
If `truncate` is sufficiently arbitrary, we could use `make_file()` in `lib` with `contents='arbitrary'`. Otherwise, `contents='random'` will use `/dev/urandom` and `dd` to generate a file with random contents (might be slow when entropy is low). Thoughts?

Reply:
I am not sure I understand.  I did not want to use up entropy which is why I used a library that only uses pseudo-random numbers.  This is also why I didn't use `contents='random'`.

Using `contents='arbitrary'` simply truncates a file which is not sufficient for my needs.

Reply:
Doing some testing, it looks like this might actually use some entropy.  I'm not sure why as hardcoding the seed and then running through the sequence I get the same results each time.

Creating a 1G file, the entropy goes down slightly but not significantly.  Either way, I can rewrite this.

Reply:
We discussed this.   The random library with a hardcoded seed does not use up entropy which is what we want.

Using `contents='arbitrary'` simply truncates a file which is not sufficient for my needs and `contents='random'` is more than I need as I don't need true randomness.

What I need is to make sure that each 1k bytes is unique.  This should be adequate to catch an algorithmic issue within the plugin while sending or receiving data.

Reply:
Yep, sounds good. We could add a new option like `contents='pseudorandom'` :p